STAC-24630: Add stackgraph backup v2 documentation [DRAFT] (#360)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/data-management/backup_restore/backup_enable.adoc

Reviewer: this is a confusing set of links

      schedule: '0 0 1 1 1970'  # January 1, 1970 (epoch start)
----

=== Enabling/disabling v1 and v2 backups

> As of {next-release-version} we make backups in two versions, v1 and v2. The old v1 system is battle-tested but slow, the v2 system is fast and feature-rich but has to prove itself, hence both systems are active by default. If for some reason one of the systems should be disabled, this can be done through `backup.stackGraph.scheduled.implementation` which can be `v1`, `v2` or `all`. The default is `all`.

Reviewer 2: Style + content on this paragraph:

- First-person voice: "we make backups" — the rest of the docs are third-person. Suggest: "As of {next-release-version}, backups are made in two versions, v1 and v2."
- Marketing/anthropomorphizing: "battle-tested", "fast and feature-rich but has to prove itself". Suggest something neutral like: "v1 is the established but slower system; v2 is faster and adds incremental backups but is newer. By default both run so v2 can be validated alongside v1."
- **Content mismatch:** this says the default is `all`, but helm-charts-internal#100 sets `backup.stackGraph.scheduled.implementation` default to `v1` (only v1 runs by default; v2 is suspended). The docs and chart need to agree — either change the chart default to `all` or update this to say v1-only by default.

Reviewer 2: Ignore `Content mismatch: this says the default is all, but helm-charts-internal#100 sets backup.stackGraph.scheduled.implementation default to v1 (only v1 runs by default; v2 is suspended). The docs and chart need to agree — either change the chart default to all or update this to say v1-only by default.`. The AI missed you comment in the helm-charts pr....

Author: I will now merge helm with v1 but move to 'all' when i release this feature. I will be testing first on noghtly while v2 is disabled for end-users

Author: applied suggestions

> As of {next-release-version}, backups are made in two versions, v1 and v2. The old v1 is the established but slower system; v2 is faster and adds incremental backups but is newer. By default both run so v2 can be used while having v1 to fall back on. If for some reason one of the systems should be disabled, this can be done through `backup.stackGraph.scheduled.implementation` which can be `v1`, `v2` or `all`. The default is `all`.

Reviewer 3: ```suggestion
As of {next-release-version}, backups are made in two versions, v1 and v2. v1 is the older, established but slower system whereas v2 is the newer, faster system that adds incremental backups. Since both systems run, by default, v2 can be primarily used while having v1 to fall back on. If, for some reason, one of the systems should be disabled, this can be done through `backup.stackGraph.scheduled.implementation` which can be `v1`, `v2` or `all`. The default is `all`.
```
"this can be done through" - What is 'this' here? Is it "backup can be performed"?
